### PR TITLE
Generate only built libs in config.cmake file

### PR DESCRIPTION
**Pull request by efernandez**
@afakihcpr @mikepurvis @tspicer01 

`BUILT_LIBRARIES` would take the values `feature;geometry;sensors;utils;sensorstream` if `BUILD_SENSORSTREAM` is `ON`, `feature;geometry;sensors;utils` otherwise.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -153,13 +153,18 @@ SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -std=gnu++0x -O0 -g3 -ggdb3
...
+SET(BUILT_LIBRARIES feature geometry sensors utils)
 
 ADD_SUBDIRECTORY(geometry)
 ADD_SUBDIRECTORY(sensors)
 
 IF(BUILD_SENSORSTREAM)
         ADD_SUBDIRECTORY(sensorstream)
+        LIST(APPEND BUILT_LIBRARIES sensorstream)
 ENDIF(BUILD_SENSORSTREAM)
+LIST(APPEND BUILT_LIBRARIES sensorstream)
```
Comment (afakihcpr) [inline src/CMakeLists.txt]: You forgot to remove this.

Comment (efernandez) [reply src/CMakeLists.txt]: Thanks. Removed!

**Comment by efernandez** [comment]
I've done this change wrt to original PR, since it seems that `FORCE` it's not needed at all:

``` diff
-SET(BUILT_LIBRARIES ${BUILT_LIBRARIES} CACHE INTERNAL "Built Libraries" FORCE)
+SET(BUILT_LIBRARIES ${BUILT_LIBRARIES} CACHE INTERNAL "Built Libraries")
```

**Comment by afakihcpr** [comment]
:+1:

**Comment by efernandez** [comment]
Tested with:

``` bash
$ catkin build --cmake-args -DBUILD_SENSORSTREAM=ON -- flirtlib
$  cat ./build/flirtlib/FLIRTConfig.cmake
# - Config file for FLIRT
#
# It defines the following variables:
#  FLIRT_INCLUDE_DIRS - Include directories for FLIRT
#  FLIRT_LIBRARIES    - Libraries for FLIRT

set(FLIRT_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR}/../../../include)

foreach(lib feature;geometry;sensors;utils;sensorstream)
  list(APPEND FLIRT_LIBRARIES ${CMAKE_CURRENT_LIST_DIR}/../../../lib/libflirtlib_${lib}.so)
endforeach()
```

and

``` bash
$ catkin build --cmake-args -DBUILD_SENSORSTREAM=OFF -- flirtlib
$ cat ./build/flirtlib/FLIRTConfig.cmake
# - Config file for FLIRT
#
# It defines the following variables:
#  FLIRT_INCLUDE_DIRS - Include directories for FLIRT
#  FLIRT_LIBRARIES    - Libraries for FLIRT

set(FLIRT_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR}/../../../include)

foreach(lib feature;geometry;sensors;utils)
  list(APPEND FLIRT_LIBRARIES ${CMAKE_CURRENT_LIST_DIR}/../../../lib/libflirtlib_${lib}.so)
endforeach()
```

**Comment by mikepurvis** [comment]
LGTM

**Pull request merged**